### ISSUE TITLE
Hiding a minimized window in Mate breaks App

### ISSUE THREAD

**Issue by averissimo** [electron/electron#8525]
* Electron version: 1.4.15
* Operating system: Ubuntu (first reported in Mate)

### Expected behavior

Tray icon actions remain available and functional

### Actual behavior

App becomes unresponsive without any error

### How to reproduce

```
git clone https://github.com/averissimo/electron-boilerplate
cd electron-boilerplate
npm install
npm start
# minimize the window
# left click on the tray icon and click on 'hide' action
# left click on the tray icon and click on 'show' action
```

The actions 'hide' and 'show' work perfectly if the window is not minimized

This was first reported in https://github.com/yakyak/yakyak/issues/563 and can also be reproduced in YakYak app.

ps. If before calling `mainWindow.hide()` we call `mainWindow.restore()` then the window has a weird behaviour with always on top frameless window. need to hide and show again for it to return to normal

**Comment by smkent** [comment]
As I just mentioned in yakyak/yakyak#563, I noticed that Slack (which also uses electron) on my work computer also exhibits this same problem.

**Comment by gch1p** [comment]
I can confirm the same in XFCE.

**Comment by deepak1556** [comment]
Sorry, I am unable to verify this since I don't have access to mate. Someone else from the team will take a look into this.

**Comment by zcbenz** [comment]
Can you try whether v1.7.x still has this problem? I tried it on Ubuntu and it worked fine, but I didn't have Mate.

**Comment by ewnd9** [comment]
Upgrading from `v1.4.15` to `v1.6.11` (current `@latest`) solved it for me, I'm on Ubuntu 16.04

**Comment by codebytere** [comment]
Closing due to abandonment.
